client/components/RelatedProducts/RelatedProducts.jsx

```diff
@@ -25,42 +29,52 @@ const RelatedProducts = ({
 
   useEffect(() => {
     getRelatedIds(productId)
-    .then(relatedIdsResult => {
-      const uniqueRelatedIdsSet = new Set(relatedIdsResult);
-      const uniqueRelatedIds = [...uniqueRelatedIdsSet]
+      .then(relatedIdsResult => {
+        const uniqueRelatedIdsSet = new Set(relatedIdsResult);
+        const uniqueRelatedIds = [...uniqueRelatedIdsSet]
 
-      const productsIdGetReq = uniqueRelatedIds.map(id => {
-        return axios.get(`https://app-hrsei-api.herokuapp.com/api/fec2/hratx/products/${id}`);
-      })
+        const productsIdGetReq = uniqueRelatedIds.map(id => {
+          return axios.get(`https://app-hrsei-api.herokuapp.com/api/fec2/hratx/products/${id}`);
+        })
 
-      const stylesGetReq = uniqueRelatedIds.map(id => {
-        return axios.get(`https://app-hrsei-api.herokuapp.com/api/fec2/hratx/products/${id}/styles`);
-      })
+        const stylesGetReq = uniqueRelatedIds.map(id => {
+          return axios.get(`https://app-hrsei-api.herokuapp.com/api/fec2/hratx/products/${id}/styles`);
+        })
 
-      const ratingsGetReq = uniqueRelatedIds.map(id => {
-        return axios.get(`https://app-hrsei-api.herokuapp.com/api/fec2/hratx/reviews/meta?product_id=${id}`);
+        const ratingsGetReq = uniqueRelatedIds.map(id => {
+          return axios.get(`https://app-hrsei-api.herokuapp.com/api/fec2/hratx/reviews/meta?product_id=${id}`);
+        })
+
+        return Promise.all([...productsIdGetReq, ...stylesGetReq, ...ratingsGetReq]);
+      })
+      .then(allResponses => allResponses.map(response => response.data))
+      .then(productCardData => {
+        //consolidates each product into one object within the combinedResults array:
+        const firstSplitIndex = productCardData.length / 3
+        const secondSplitIndex = firstSplitIndex * 2;
+        const productsIdResult = productCardData.slice(0, firstSplitIndex);
+        const stylesResult = productCardData.slice(firstSplitIndex, secondSplitIndex);
+        const ratingsResult = productCardData.slice(secondSplitIndex)
+        const combinedResults = [];
+
+        for (let i = 0; i < productsIdResult.length; i++) {
+          let temp = { ...productsIdResult[i], ...stylesResult[i], ...ratingsResult[i] };
+          combinedResults.push(temp);
+        }
+
+        setRelatedProductsData(combinedResults)
       })
+  }, [productId]);
```

Comment:
Thank you so much for your note that I don't have to review this code.  Glad I don't have to review this haha :)